scripts/js/basic_usage.js

```diff
@@ -28,14 +28,49 @@ async function getBlockWithRetry(web3){
 
       i++;
       if (i === 10){
-        throw new Error('Failed to connect to Infura over websockets after 10 tries');
+        if (provider === 'infura') {
+          throw new Error('Failed to connect to Infura over websockets after 10 tries');
+        } else if (provider === 'alchemy') {
+          throw new Error('Failed to connect to Alchemy over websockets after 10 tries');
+        }
       }
 
     }
   }
   return block;
 }
 
+async function tryMultipleProviders(infuraProvider) {
+  try {
+    const web3 = new Web3(infuraProvider);
+    const block = await getBlockWithRetry(web3, 'infura');
+    return [
+      web3,
+      block
+    ];
+  } catch (error) {
+    if (infuraProvider.includes("wss")) {
+      // Fetched provider from ethers.js https://github.com/ethers-io/ethers.js/blob/50d712d6afc927aaee68b482157598554a45d403/packages/providers/src.ts/alchemy-provider.ts#L21
+      const alchemyProvider = "wss://eth-mainnet.alchemyapi.io/v2/_gg7wSSi0KMBsdKnGVfHDueq6xMB9EkC";
```

Comment:
@jdevcs I think that the ethers.js maintainer just opened the API keys to be used like public endpoints

https://github.com/ethers-io/ethers.js/blob/540a670f94829265be57a8b3f84b33308b37fcbf/docs.wrm/api-keys.wrm#L13

```
These API keys are provided as a community resource by the backend services
for low-traffic projects and for early prototyping.

Since these API keys are shared by all users (that have not acquired their
own API key), they are aggressively throttled which means retries occur more
frequently and the responses are slower.

It is **highly recommended** that you sign up for a free API key from each service for their
free tier, which (depending on the service) includes many advantages:
```

So, I think using the API key as a fallback test would not be a problem.

Reply:
I think our proj might not qualify for: " low-traffic projects and for **early prototyping** " and there is also **aggressively throttled** constraint.